### PR TITLE
docs: handle beta GW apis for 2.6.x KIC

### PR DESCRIPTION
**Pull request by shaneutt**
### Summary

Update the Gateway API guide for `v0.5.0` in KIC `v2.6.0` to use beta resources and indicate that the the integration is now considered beta.

### Reason

Resolves https://github.com/Kong/docs.konghq.com/issues/4123

### Testing

Manually tested the documentation to verify the correct result.

### CODE REVIEW

**Comment by lena-larionova** [comment]
Preview: https://deploy-preview-4238--kongdocs.netlify.app/kubernetes-ingress-controller/latest/

**Comment by rainest** [comment]
This should be do not merge still, correct? 2.6 isn't out yet.

Does the page metadata also support version-based if template blocks? This currently changes https://deploy-preview-4238--kongdocs.netlify.app/kubernetes-ingress-controller/2.5.x/guides/using-gateway-api/ and https://deploy-preview-4238--kongdocs.netlify.app/kubernetes-ingress-controller/2.4.x/guides/using-gateway-api/ to beta retroactively.

We're also adding new pages that will need annotation updates also:

https://github.com/Kong/docs.konghq.com/pull/4231/files (missing alpha/beta annotations)
https://github.com/Kong/docs.konghq.com/pull/4230/files (pending docs team review)

**Comment by lena-larionova** [comment]
> Does the page metadata also support version-based if template blocks? This currently changes https://deploy-preview-4238--kongdocs.netlify.app/kubernetes-ingress-controller/2.5.x/guides/using-gateway-api/ and https://deploy-preview-4238--kongdocs.netlify.app/kubernetes-ingress-controller/2.4.x/guides/using-gateway-api/ to beta retroactively.

Hmm. No, page metadata doesn't support that. @mheap do you have any ideas for handling this? Right now, the thought that comes to mind is splitting the source files. The alpha/beta/GA use case might be a good one for that.

File: src/kubernetes-ingress-controller/guides/using-gateway-api.md [hunk 1]
```diff
@@ -1,33 +1,39 @@
...
 title: Using Gateway API
-alpha: true
 ---
 
+{% if_version gte: 2.6.x %}
+This guide is considered `beta` maturity: it is not intended for use in production systems.
+{% endif_version %}
+{% if_version lte: 2.5.x %}
+This guide is considered `alpha` maturity: it is not intended for use in production systems.
+{% endif_version %}
```
Comment (lena-larionova) [inline src/kubernetes-ingress-controller/guides/using-gateway-api.md:10]:
```suggestion
{% if_version gte: 2.6.x %}
{:.note}
> This guide is considered `beta` maturity. It is not intended for use in production systems.
{% endif_version %}
{% if_version lte: 2.5.x %}
{:.note}
> This guide is considered `alpha` maturity. It is not intended for use in production systems.
{% endif_version %}
```

**Review by lena-larionova — APPROVED** [review]
LGTM! Adding the banners manually did the job for now. We'll have to look into better ways of handling metadata in the future, but that's a doc team to-do already.

Feel free to merge whenever, since this is going into a release branch anyway.

**Pull request merged**